text/0000-compiler-team-trusted-contrib-maintainer.md

```diff
@@ -0,0 +1,381 @@
+that the compiler team is responsible for (with all of the expected caveats
+for vacation time, mental health breaks, etc) - listed as [responsibilities]
+[responsibilities] below. However, not all contributors need to participate
+in these responsibilities to an equal degree. Contributors should participate
+in these tasks to the degree that they are able - volunteers are not expected
+to participate as much as those employed to work on the compiler, for example.
+It is the responsibility of the compiler team leads to ensure that the ongoing
+maintenance tasks of the team can be completed sustainably.
+
+Team members have all of the same permissions and access as trusted
+contributors. Like trusted contributors, team members are considered members of
+the Rust project as a whole and are expected to follow the spirit of the [Code
+of Conduct][coc].
+
+Like trusted contributors, after inactivity for longer than a year, a
+contributor will be moved to alumni status. Members who are no longer able to
+help maintain the compiler but otherwise wish to continue contributing to the
+compiler can also be moved to alumni status and retain their trusted contributor
+status. Alumni can ask to be reinstated in future.
+
+[coc]: https://www.rust-lang.org/policies/code-of-conduct
+
+Responsibilities
+----------------
+[responsibilities]: #responsibilities
+
+There are various responsibilities that a team member could take on to help the
+team. All team members must participate in at least one activity.
+
+Team members can get involved in any of these by contacting the team leads, by
+asking team members currently involved in these responsibilities, or by asking
+in any venue where these responsibilities are conducted (e.g. a Zulip stream).
+
+- Final comment period (FCP) reviewer
+    - Final comment periods are the process by which the team signs-off on a
+      change before it is made, like stabilizing a feature.
+
+      FCPs have always required whole team to sign-off, but this doesn't scale
+      as the team grows. As described above, this acts as a disincentive for the
+      team to grow. Furthermore, not all FCPs are relevant to all team members and a
+      diffusion of responsibility means that most team members just sanity-check and
+      then sign-off. This isn't ideal, as it doesn't guarantee that someone on the
+      team has thoroughly considered a FCP.
+
+      Instead, have FCPs require sign-off from team members who opt-in to being
+      an "FCP reviewer", with the expectation that they will spend time reviewing
+      an FCP thoroughly. FCP reviewers should also consider reaching out to relevant
+      domain experts and soliciting their opinions whenever possible. Any team or
+      project member can raise concerns with an FCP, which will be considered by the
+      FCP reviewers.
+      
+      To function effectively, it is recommended that there be 4 - 8 FCP
```

Comment:
By what mechanism do we prevent growth beyond the desired range? Like, what if someone wants to be on FCP duty, but we already got 8 folk on the list?

Reply:
Does a 4-8 limit actually matter in practice if all the members are responding "timely". We'd still have the problem of "FCPs span a wide variety of required knowledges and we dont want to encourage mindlessly checking a box because its not your area" but honestly maybe that's fine and if we are explicitly sayign to Not Do This we'd encourage growing the amount of documentation we have for areas so that people can more accurately evaluate fcps

Reply:
"by signing up for FCPs you opt-in to start getting pinged after a week if you haven't reacted to the FCP. Contributors are encouraged to ping FCP folk if there are no more outstanding concerns."

Reply:
I've added some wording here - we don't want a hard upper limit, the goal is just to have a reasonable amount of diversity of opinion in the FCP list; make sure that everything gets a thorough look; and doesn't take too long. We want the FCP list to be populated by people who are committing to never mindlessly checking their box, they're on the list because they're interested in engaging with all of the FCPs we get. Team members who aren't on the FCP list and have an opinion can still register concerns and things like that to be heard.

Reply:
> "by signing up for FCPs you opt-in to start getting pinged after a week if you haven't reacted to the FCP. Contributors are encouraged to ping FCP folk if there are no more outstanding concerns."

GitHub hadn't shown me this comment when I was posting mine, I can change the wording to be closer to this if that's preferred.

Reply:
I think specifying expectations for how frequently you're going to be pinged for reviews if you are an FCP reviewer is a good idea!

As David said, the upper limit is really just a suggestion, after ~8 people just collecting reviews from everyone in the group tends to take a while even when no major issues are raised. It's certainly possible that a larger group could also work if the FCP reviewers respond in a very timely manner.

Reply:
I think the issue wouldn't be that there are too many people, but that some people on the list aren't reviewing promptly.

If we had twenty people on the list and everyone reviewed within a day, I think that would be okay, it's about making sure things move forward and setting expectations about promptness in FCP participation, rather than the number of people (unless maybe it meant that there was always _some_ objection so nothing would progress - but that's a separate problem that we can deal with if we run into it).